projects/components/src/input/input.component.ts

```diff
@@ -66,4 +66,8 @@ export class InputComponent<T extends string | number> implements OnChanges {
         return value as T | undefined;
     }
   }
+
+  public getStyleClasses(): string[] {
```

Comment:
Why not create a style property that we update whenever `appearance`  or `disabled` changes?